src/Tmds.Ssh/SshConnection.cs

```diff
@@ -0,0 +1,40 @@
+﻿// This file is part of Tmds.Ssh which is released under LGPL-3.0.
+// See file LICENSE for full license details.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Tmds.Ssh
+{
+    public sealed class SshConnection : IDisposable
+    {
+        private readonly SshConnectionSettings _settings;
```

Comment:
It might be good to be able to retrieve a connection object's settings after it has been created. Maybe for classification purposes (who it is connected to, what port is it using, what crypto protocols does it use, etc.), or to possibly duplicate them after the fact (like 'restarting' a previously connected session).

So maybe a getter property here would be appropriate?

Reply:
> It might be good to be able to retrieve a connection object's settings after it has been created.

I'm experimenting with a certan design here which separates these concerns:
- providing settings that control the connection behavior -> no getter for settings
- calling connect -> no getter to know if connectasync has been called

We can add them later if we want to. Removing them later is more difficult.

> Maybe for classification purposes (who it is connected to, what port is it using, what crypto protocols does it use, etc.), or to possibly duplicate them after the fact (like 'restarting' a previously connected session).

Yes, providing such info would be interesting. We should put them on a separate object, like `ConnectionInfo`.